### ISSUE TITLE
Bug: [no-unused-vars] false positive if name is type/value merged and exported

### ISSUE THREAD

**Issue by bradzacher** [typescript-eslint/typescript-eslint#6188]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=4.9.3&jsx=true&sourceType=module&showAST=es&code=C4TwDgpgBAZg9nKBeKBvAvgbgFAQB5hwBOwUAxnAHYDOp8iKAFAIZEDmAXLAgJTIB8abFChEIwAK5FKUVmxzpsQA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1tidmUQAmtAG4BDaKgyRE0aB2iRwYAL4hlQA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA

### Repro Code

```typescript
type foo = {};
export const foo = (arg: foo) => {
  return arg;
}
```

### ESLint Config

```javascript
module.exports = {
  "rules": {
    "@typescript-eslint/no-unused-vars": "error"
  }
}
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "strictNullChecks": true
  }
}
```

### Expected Result

no error because the type `foo` is used and the value `foo` is exported

### Actual Result

```
'foo' is assigned a value but never used. 2:14 - 2:17
```

**Comment by mithodin** [comment]
Also affected by this, we use the following quite often for React components:
```tsx
interface Test {
    test: string;
}

export const Test = ({ test }: Test) => <>{test}</>;
```

**Comment by Maxim-Mazurok** [comment]
Having the same issue with `ToStringAble` even though it's definitely used:

```ts
interface ToStringAble {
  toString(): string;
}
export interface EqualAble {
  equals(other: EqualAble): boolean;
}
export const ToStringAble = (object: ToStringAble): EqualAble => ({
  equals: (other: ToStringAble) => object.toString() === other.toString(),
});
```

**Comment by bradzacher** [comment]
The bug is in this function here:
https://github.com/typescript-eslint/typescript-eslint/blob/8ee0cbb0818392cc818b1c254dcec6e879f79a35/packages/eslint-plugin/src/util/collectUnusedVariables.ts#L425-L440

specifically that it only considers if the FIRST declaration is exported.
So if the second declaration is exported, the logic doesn't pick that up.

**Comment by Maxim-Mazurok** [comment]
Thanks for suggestion @bradzacher , opened PR: https://github.com/typescript-eslint/typescript-eslint/pull/6873
